hdf_compass/bag_model/model.py

```diff
@@ -282,18 +286,158 @@ def is_plottable(self):
         return True
 
 
-class BAGElevation(compass_model.Array):
+class BAGElevationArray(compass_model.Array):
+    """ Represents a BAG elevation. """
+    class_kind = "BAG Elevation [array]"
+
+    @staticmethod
+    def can_handle(store, key):
+        return (key == "/BAG_root/elevation") and (key in store) and (isinstance(store.f[key], h5py.Dataset))
+
+    def __init__(self, store, key):
+        self._store = store
+        self._key = key
+        self._dset = store.f.elevation(mask_nan=True)
+
+    @property
+    def key(self):
+        return self._key
+
+    @property
+    def store(self):
+        return self._store
+
+    @property
+    def display_name(self):
+        return pp.basename(self.key)
+
+    @property
+    def description(self):
+        return 'Dataset "%s"' % (self.display_name,)
+
+    @property
+    def shape(self):
+        return self._dset.shape
+
+    @property
+    def dtype(self):
+        return self._dset.dtype
+
+    def __getitem__(self, args):
+        return self._dset[args]
+
+
+class BAGElevationGeoArray(compass_model.GeoArray):
+    """ Represents a BAG elevation. """
+    class_kind = "BAG Elevation [geo array]"
+
+    @staticmethod
+    def can_handle(store, key):
+        return (key == "/BAG_root/elevation") and (key in store) and (isinstance(store.f[key], h5py.Dataset))
+
+    def __init__(self, store, key):
+        self._store = store
+        self._key = key
+        self._dset = store.f.elevation(mask_nan=True)
+        self._meta = store.f.populate_metadata()
+
+    @property
+    def key(self):
+        return self._key
+
+    @property
+    def store(self):
+        return self._store
+
+    @property
+    def display_name(self):
+        return pp.basename(self.key)
+
+    @property
+    def description(self):
+        return 'Dataset "%s"' % (self.display_name,)
+
+    @property
+    def shape(self):
+        return self._dset.shape
+
+    @property
+    def dtype(self):
+        return self._dset.dtype
+
+    @property
+    def extent(self):
+        """ Geographic extent as a tuple: (x_min, x_max, y_min, y_max) """
+        return self._meta.geo_extent()
+
+    def __getitem__(self, args):
+        return self._dset[args]
+
+
+class BAGElevation(compass_model.GeoSurface):
     """ Represents a BAG elevation. """
     class_kind = "BAG Elevation"
 
     @staticmethod
     def can_handle(store, key):
+        import matplotlib
+        # for GeoSurface we are using a matplotlib function present after 1.5.x
+        plt_maj, plt_min, _ = matplotlib.__version__.split('.')
+        if (int(plt_maj) == 1) and (int(plt_min) < 5):
```

Comment:
Correct.. but I have added this additional check since the app is not always installed with ``setup.py``. It was only recently added.